evmcore/state_processor_test.go

```diff
@@ -258,6 +257,67 @@ func TestProcess_TracksParentBlockHashIfPragueIsEnabled(t *testing.T) {
 	}
 }
 
+func TestProcess_FailingTransactionAreSkippedButTheBlockIsNotTerminated(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	state := state.NewMockStateDB(ctrl)
+
+	chainConfig := params.ChainConfig{}
+	chain := NewMockDummyChain(ctrl)
+	processor := NewStateProcessor(&chainConfig, chain)
+
+	block := &EvmBlock{
+		EvmHeader: EvmHeader{
+			Number:   big.NewInt(1),
+			GasLimit: 100_000,
+		},
+		Transactions: []*types.Transaction{
+			// This transaction will fail due to an invalid signature.
+			types.NewTx(&types.LegacyTx{
+				Nonce:    0,
+				To:       &common.Address{},
+				Gas:      21_000,
+				GasPrice: big.NewInt(1),
+				V:        big.NewInt(1), // invalid signature
+			}),
+			// Valid transaction that will succeed.
+			types.NewTx(&types.LegacyTx{
+				Nonce:    0,
+				To:       &common.Address{},
+				Gas:      21_000,
+				GasPrice: big.NewInt(1),
+			}),
+		},
+	}
+
+	// Mock the state database interactions for passing transaction.
+	any := gomock.Any()
+	state.EXPECT().SetTxContext(any, any).Times(1)
+	state.EXPECT().GetBalance(any).Return(uint256.NewInt(1000000)).Times(1)
+	state.EXPECT().SubBalance(any, any, any).Times(2)
+	state.EXPECT().Prepare(any, any, any, any, any, any).Times(1)
+	state.EXPECT().GetNonce(any).Return(uint64(0)).Times(1)
+	state.EXPECT().SetNonce(any, any, any).Times(1)
+	state.EXPECT().GetCode(any).Return(nil).Times(2)
+	state.EXPECT().Snapshot().Return(0).Times(1)
+	state.EXPECT().Exist(any).Return(true).Times(1)
+	state.EXPECT().AddBalance(any, any, any).Times(3)
+	state.EXPECT().GetRefund().Return(uint64(0)).Times(2)
+	state.EXPECT().GetLogs(any, any).Return([]*types.Log{})
+	state.EXPECT().EndTransaction().Times(1)
+	state.EXPECT().TxIndex().Return(0).Times(1)
+
+	// Process the block
+	usedGas := new(uint64)
+	receipts, logs, skipped := processor.Process(block, state, vm.Config{}, usedGas, nil)
+
+	require.Len(t, receipts, 2)
+	require.Nil(t, receipts[0])
+	require.NotNil(t, receipts[1])
+
+	require.Empty(t, logs)
+	require.Empty(t, skipped)
```

Comment:
looking at this, it gives the impression that there this change introduces a somewhat hidden semantic, which is "an erroneus transaction - which has no recepit, nor is counted as skipped, but it also does not produce a failed receipt", was this the intent?